### DISCUSSION TITLE
Where to put plugins config?

### DISCUSSION THREAD

**Discussion by jksjaz** [apollographql/router#1909] [Q&A] [answered]
While working with the helm chart of the router, where should I put the plugins section?

I am trying to add the plugin for redacted errors as I want to show the errors to the client.
As the documentation mentions, it needs to be in the router.yaml, how do we suppose to add that file in the helm?
If the assumption is that I should use the values.yaml, please share a configuration example showing where exactly is the plugins section suppose to be.

Note: I have tried different combinations already like putting the following block under these sections: router.configuration, router.plugins, plugins (at root)

```
plugins:
  include_subgraph_errors:
    all: true
```

**Comment by garypen** [comment, marked as answer]
Apollo plugins have their configuration defined at the root level.

So, if running a router natively, your configuration file should include:
```
include_subgraph_errors:
  all: true
```

To deploy a router with this configuration to k8s using our helm chart, your values.yaml should include:
```
router:
  configuration:
    include_subgraph_errors:
      all: true
```
plus whatever other configuration you desire.

Our docs are helpful:

https://www.apollographql.com/docs/router/configuration/subgraph-error-inclusion
https://github.com/apollographql/router/tree/dev/helm/chart/router

**Reply by jksjaz** [reply]
@garypen any suggestions on how to use it with terraform?

If I `set` `router.configuration.include_subgraph_errors.all: true` it doesn't work and it also complains about not being a valid configuration using the below shows `values` section

```
resource "helm_release" "router-dev" {
  name       = "apollo-router-dev"  
  chart      = "oci://ghcr.io/apollographql/helm-charts/router"
  values     = ["${file("router/values.yaml")}"]

  set {
    name  = "router.configuration.telemetry.metrics.prometheus.enabled"
    value = true
  }

  set {
    name  = "managedFederation.apiKey"
    value = var.APOLLO_KEY
  }

  set {
    name  = "managedFederation.graphRef"
    value = "TEST@dev"
  }
}
```

**Reply by garypen** [reply]
I'm lucky enough not to be using terraform actively at the moment, so I'm not sure my suggestions will be very helpful...

I think the correct syntax for that set should be:

```
  set {
    name  = "router.configuration.include_subgraph_errors.all"
    value = true
  }
````
but, I may well be wrong.
